bot.go

```diff
@@ -103,107 +96,103 @@ func (b *Bot) Join(s *discordgo.Session, m *discordgo.MessageCreate, name string
-					var msg strings.Builder
-					msg.Grow(32)
-					for player := range b.games[g].Players {
-						fmt.Fprintf(&msg, "%s | ", player)
-					}
-					s.ChannelMessageSend(m.ChannelID, msg.String())
+					*game.BlueCaptain = player
+					game.Blue[player] = true
 				}
-				b.games[g].mutex.Unlock()
-			} else {
-				b.games[g].mutex.Unlock()
-				b.List(s, m, name)
 			}
+			b.teamsSelected(s, m, *gameID)
+		} else {
+			players := game.Players
+			*game.RedCaptain = game.RandPlayer()
+			game.Red[*game.RedCaptain] = true
+			delete(players, *game.RedCaptain)
+			*game.BlueCaptain = game.RandPlayer()
+			game.Blue[*game.BlueCaptain] = true
+			delete(players, *game.BlueCaptain)
+			s.ChannelMessageSend(m.ChannelID, fmt.Sprintf("%s is captain for red team, %s is captain for blue team", *game.RedCaptain, *game.BlueCaptain))
+			s.ChannelMessageSend(m.ChannelID, fmt.Sprintf("%s to pick", *game.RedCaptain))
+			s.ChannelMessageSend(m.ChannelID, game.BuildPlayerList())
 		}
+		game.mutex.Unlock()
+	} else {
+		game.mutex.Unlock()
+		b.List(s, m, name)
 	}
 }
 
 func (b *Bot) Pick(s *discordgo.Session, m *discordgo.MessageCreate, modName string, playerName string) {
-	if c, ok := b.channels[m.ChannelID]; ok {
-		g := GameIdentifier{m.ChannelID, modName}
-		if _, ok := b.games[g]; ok {
-			mod := c.Mods[modName]
-			// Game found, check whether it's filled.
-			if len(b.games[g].Players)+len(b.games[g].Red)+len(b.games[g].Blue) != mod.MaxPlayers {
-				return
-			}
-			if _, ok := b.games[g].Players[playerName]; !ok {
-				return
-			}
-			// 0 is red, 1 is blue
-			pickColor := 0
-			pickedCount := len(b.games[g].Red) + len(b.games[g].Blue)
-			if pickedCount > 0 && ((pickedCount-1)/2)%2 == 1 {
-				pickColor = 1
-			}
-			if pickColor == 0 && (*b.games[g].RedCaptain == m.Message.Author.Username || m.Message.Author.Username == "hyperreal") {
-				b.games[g].Red[playerName] = true
-				delete(b.games[g].Players, playerName)
-			}
-			if pickColor == 1 && (*b.games[g].BlueCaptain == m.Message.Author.Username || m.Message.Author.Username == "hyperreal") {
-				b.games[g].Blue[playerName] = true
-				delete(b.games[g].Players, playerName)
-			}
-			if len(b.games[g].Players) == 1 {
-				if len(b.games[g].Red) < len(b.games[g].Blue) {
-					b.games[g].Red[playerName] = true
-				} else {
-					b.games[g].Blue[playerName] = true
-				}
-				b.teamsSelected(s, m, g)
-			}
-		}
+	gameID, mod := b.GameInfo(m.ChannelID, modName)
+	if gameID == nil || mod == nil {
+		return
 	}
-}
 
-func RandPlayer(players map[string]PlayerMetadata) string {
-	keys := reflect.ValueOf(players).MapKeys()
-	return keys[rand.Intn(len(keys))].String()
+	if _, ok := b.games[*gameID]; !ok {
+		return
+	}
+
+	game := b.games[*gameID]
+
+	if !game.IsPickingTeams(mod) {
+		return
+	}
+
+	if !game.HasPlayer(playerName) {
+		return
+	}
+
+	// 0 is red, 1 is blue
+	pickColor := 0
+	pickedCount := len(game.Red) + len(game.Blue)
```

Comment:
If you're refactoring, you could also refactor this logic into its own function, i.e. return color of captain picking based on the count of players in each team, so you can print out who is the next person to pick at the end of this function.

Reply:
I am working on team picking on a separate branch atm. Captain command, countdown timer, and all that